dask/array/routines.py

```diff
@@ -290,6 +290,9 @@ def matmul(a, b):
     return out
 
 
+moveaxis = np.moveaxis
```

Comment:
```suggestion
def numpy_implementation(func, *args, **kwargs):
    if hasattr(func, '_implementation'):
        return func._implementation(*args, **kwargs)
    else:
        return func(*args, **kwargs)

moveaxis = partial(numpy_implementation, np.moveaxis)
```

Reply:
Probably `numpy_implementation` could go somewhere more general, such as `dask/utils.py`.

Reply:
Seems like a good idea. Could add to [dask/array/numpy_compat.py]( https://github.com/dask/dask/blob/master/dask/array/numpy_compat.py ).

Reply:
Please don't use `_implementation`. This is not a public API exposed by NumPy, as indicated by the name starting with an underscore. (If this isn't clear from the NEP, we should fix that.)

It would be more maintainable to copy the implementations from NumPy. These are not complicated functions, so I really don't think there is much that could go wrong with copying their code.

Reply:
Is there an alternative to copying the implementation? These are not the only cases where this will be necessary. In CuPy as well we need this for some functions that are currently aliases, I've submitted a PR for those https://github.com/cupy/cupy/pull/2249.

Granted, I was very inattentive to `_implementation` being non-public.